contracts/dynamic-caller-contract/src/contract.rs

```diff
@@ -201,6 +216,29 @@ pub fn try_validate_interface_err(deps: Deps, _env: Env) -> Result<Response, Con
     Ok(Response::default())
 }
 
+// check the caller_address works
+pub fn try_call_caller_address_of(
+    deps: Deps,
+    _env: Env,
+    target: Addr,
+) -> Result<Response, ContractError> {
+    let address: Addr = from_slice(
+        &deps
+            .storage
+            .get(b"dynamic_callee_contract")
+            .ok_or_else(|| ContractError::Storage("cannot get callee address".to_string()))?,
+    )?;
+    let contract = CalleeContract {
+        address: address.clone(),
+    };
+    let result_addr = contract.call_caller_address_of(target);
+
+    let res =
+        Response::default().add_attribute("is_as_expected", (result_addr == address).to_string());
```

Comment:
Am I correct that the comparison here is `False`?

Reply:
No, this is expected to be `True`.

Reply:
Understood, thank you.